### PR TITLE
feat: inline Snyk docker image scan

### PR DESCRIPTION
**Pull request by ernestojeda**
This PR modifies `edgeXSnyk.groovy` to allow on-demand testing of images that are pushed to a docker registry.

edgeXSnyk should:

- implement on-demand testing of docker images (pushed to nexus)
- notify distribution list of users when vulnerabilities are found
- allow ignoring of Snyk findings via the `.snyk` policy

Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/edgexfoundry/ci-management/blob/master/.github/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Added labels
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Sandbox Testing
Test Links :

See jobs under this PR:
`edgeXBuildGoApp`
https://jenkins.edgexfoundry.org/job/edgexfoundry/job/sample-service/view/change-requests/job/PR-77/

`edgeXBuildGoAppParallel`
https://jenkins.edgexfoundry.org/sandbox/view/All/job/Functional-Testing/job/eojeda-edgex-go-snyk/4/console

Example HTML Email:
image

Example Plain Text Email:
image

### CODE REVIEW

File: vars/edgeXSnyk.groovy [hunk 1]
```diff
@@ -14,26 +14,116 @@
...
+            if(sendEmail && snykRecipients) {
+                // send email to committer and secuirty group here
+                def subject = "[Snyk] Possible vulnerabilities discovered in ${env.PROJECT ? '[' + env.PROJECT + ']' : ''} scan"
+
+                if(htmlReport) {
+                    def messageBody = readFile(file: './snykReport.html')
+
+                    mail body: messageBody, subject: subject, to: snykRecipients, mimeType: 'text/html'
+                } else {
+                    def messageBody = """Possible vulnerablities have been found by Snyk in for the following build: ${env.JOB_NAME}
```
Comment (soda480) [inline vars/edgeXSnyk.groovy:101]: spelling error: `vulnerabilities`

File: vars/edgeXSnyk.groovy [hunk 2]
```diff
@@ -14,26 +14,116 @@
...
+
+    if(exitCode > 0) {
+        println "[edgeXSnyk] Exit Code: ${exitCode}"
+        // https://support.snyk.io/hc/en-us/articles/360003812578-CLI-reference
+
+        // Exit codes only applicable to snyk test
+        // Exit code 0 This means Snyk did not find vulnerabilities in your code an exited the process without failing the job.
+        // Exit code 1 This means Snyk found vulnerabilities in your code and have failed the build
+        // Exit code 2 This means Snyk exited with an error, please re-run with `-d` to see further information.
+        // Exit code 3 This means Snyk did not detect any supported projects/manifests to scan. Re-check the command or if the command should run in a different directory.
```
Comment (soda480) [inline vars/edgeXSnyk.groovy:87]: thanks for adding exit code representations

**Review by soda480 — APPROVED** [review]
Looks Great - just found a minor misspelling in the email message body

**Review by bill-mahoney — APPROVED** [review]
LGTM

**Review by cjoyv — APPROVED** [review]
Looks great

File: vars/edgeXSnyk.groovy (hunk 1 above)
Comment (ernestojeda) [reply vars/edgeXSnyk.groovy:101]: I will open a separate PR for this

**Pull request merged**